packages/opentelemetry-cloud-trace-propagator/src/CloudPropagator.ts

```diff
@@ -0,0 +1,102 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {
+  HttpTextPropagator,
+  Context,
+  SetterFunction,
+  GetterFunction,
+  TraceFlags,
+} from '@opentelemetry/api';
+import { decToHex, hexToDec } from 'hex2dec';
+import {
+  setExtractedSpanContext,
+  getParentSpanContext,
+} from '@opentelemetry/core';
+import * as crypto from 'crypto';
+
+/**
+ * This file implements propagation for the Stackdriver Trace v1 Trace Context
+ * format.
+ *
+ * The header specification is:
+ * "X-Cloud-Trace-Context: TRACE_ID/SPAN_ID;o=TRACE_TRUE"
+ * Where:
+ *  {TRACE_ID} is a 32-character hexadecimal value representing a 128-bit
+ *    number. It should be unique between your requests, unless you
+ *    intentionally want to bundle the requests together.
+ *  {SPAN_ID} is the decimal representation of the (unsigned) span ID. It
+ *    should be 0 for the first span in your trace. For subsequent requests,
+ *    set SPAN_ID to the span ID of the parent request.
+ *  {TRACE_TRUE} must be 1 to trace request. Specify 0 to not trace the request.
+ */
+
+/** Header that carries span context across Google infrastructure. */
+export const X_CLOUD_TRACE_HEADER = 'x-cloud-trace-context';
+const TRACE_ID_REGEX = /^([0-9a-f]{32})/;
+const SPAN_ID_REGEX = /\/(\d+)/;
```

Comment:
Problem of changing this to use 3 different regexes, is that now is much harder to enforce the format.

For example:
`d4cda95b652f4a1592b449d5929fda1b/7929822056569588882o=1`,
`d4cda95b652f4a1592b449d5929fda1b/foo7929822056569588882bar;o=1`

 will now be valid.